vectorstores/alloydb/options.go

```diff
@@ -19,7 +19,7 @@ type engineConfig struct {
 	database        string
 	user            string
 	password        string
-	ipType          string
+	usePrivateIP    bool
```

Comment:
nit: I would keep this with "ipType" and have WithPrivateIP set ipType = private